tests/unit-tests/graphics/test_software_cursor.cpp

```diff
@@ -194,9 +194,9 @@ TEST_F(SoftwareCursor, tolerates_being_hidden_while_being_shown)
     EXPECT_CALL(mock_input_scene, add_input_visualization(_))
         .WillOnce(Invoke([&](auto)
             {
-                cursor.hide(); // should do nothing
+                cursor.hide();
             }));
-    EXPECT_CALL(mock_input_scene, remove_input_visualization(_)).Times(0);
+    EXPECT_CALL(mock_input_scene, remove_input_visualization(_));
```

Comment:
> Also the software cursor tests where wrong.

Agreed, they were wrong. But I think they are still (differently) wrong.

IMO they are testing that "something sensible" happens `hide()` is called during `show()` or _vice versa_.

They used to test that `hide()` has no effect (after `executor.execute()`), they now test that it has effect.

```suggestion
    EXPECT_CALL(mock_input_scene, remove_input_visualization(_)).AnyNumber();
```

I don't think _these_ tests should be mandating whether `hide()` has effect here.